pubsub/azure/eventhubs/eventhubs.go

```diff
@@ -124,6 +127,16 @@ func subscribeHandler(ctx context.Context, topic string, e *eventhub.Event, hand
 	if e.ID != "" {
 		res.Metadata[sysPropMessageID] = e.ID
 	}
+	// added properties if any ( includes application properties from iot-hub)
+	if getAllProperties{
+		if e.Properties != nil && len(e.Properties) > 0{
+			for key, value := range e.Properties {
+				if str, ok := value.(string); ok {
+						res.Metadata[key] = str
```

Comment:
is it possible for these property names to conflict with other metadata names? If so this step would be replacing them...

Reply:
The custom application properties will be a dictionary of string properties that the device ( or client ) will define itself other than the ones already pre-defined under system properties. I believe that the possibility of chances of intentional conflict among them is maybe impractical. 

List of predefined system properties : [System Properties of D2C IoT Hub messages](https://learn.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-construct#system-properties-of-d2c-iot-hub-messages)